ci/Jenkinsfiles/build.groovy

```diff
@@ -203,8 +202,50 @@ def buildUnitTestStage(env) {
   }
 }
 
+def buildFrontendTestStage() {
+  return {
+    stage('Run frontend unit tests') {
+      container(DEFAULT_CONTAINER) {
+        script {
+          try {
+            setGitHubBuildStatus('utests/frontend', 'Unit tests - frontend', 'PENDING')
+            sh 'cd nuxeo-compound-documents-web && npm run test'
+            setGitHubBuildStatus('utests/frontend', 'Unit tests - frontend', 'SUCCESS')
+          } catch(err) {
+            currentBuild.result = "FAILURE"
+            setGitHubBuildStatus('utests/frontend', 'Unit tests - frontend', 'FAILURE')
+            throw err
+          }
+        }
+      }
+    }
+  }
+}
+
+String getMavenFailArgs() {
+  return (isPullRequest() && pullRequest.labels.contains('failatend')) ? '--fail-at-end' : ' '
+}
+
+void runFunctionalTests(String baseDir) {
+  try {
+    retry(2) {
+      sh "mvn ${MAVEN_ARGS} ${MAVEN_FAIL_ARGS} -f ${baseDir}/pom.xml verify"
+    }
+    findText regexp: ".*ERROR.*", fileSet: "ftests/**/log/server.log"
+  } catch(err) {
+    echo "${baseDir} functional tests error: ${err}"
+    throw err
+  } finally {
+    try {
+      archiveArtifacts allowEmptyArchive: true, artifacts: "${baseDir}/**/target/failsafe-reports/*, ${baseDir}/**/target/**/*.log, ${baseDir}/**/target/*.png, ${baseDir}**/target/cucumber-reports/*.json, ${baseDir}/**/target/*.html, ${baseDir}/**/target/**/distribution.properties, ${baseDir}/**/target/**/configuration.properties"
+    } catch (err) {
+      echo hudson.Functions.printThrowable(err)
+    }
+  }
+}
+
 String getCurrentNamespace() {
-  container('maven') {
+  container('ftests') {
```

Comment:
Shound we se DEFAULT_CONTAINER here?
```suggestion
  container(DEFAULT_CONTAINER) {
```

Reply:
No, at least not now. Left you a message on slack about this issue.

Reply:
Not sure I understand why not, because `DEFAULT_CONTAINER` is already `ftests`.